generate.go

```diff
@@ -18,9 +18,12 @@ import (
 //go:embed tmpl/* tmpl/.*
 var tmplFS embed.FS
 
-func generate(rootPath string, plan *planner.Plan) error {
+func generate(rootPath string, plan *planner.Plan, includeBuildFiles bool) error {
```

Comment:
I don't love the `bool`.

How about this:
you change `generate` to take a string array.

you add `generateNix` (which calls `generate` with `.gitignore`, `shell.nix`, `default.nix`)

you add 'generateDockerfile` (which calls `generate` with `.gitignore`, `Dockerfile`)